ls1intum/Artemis
# `Exam mode`: Allow instructors to reset an exam

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


When testing the exam mode feature, it would be handy if you could simply reset an exam: this would keep its configuration, exercise groups, exercises and registered students, but would delete all student exams, all participations, including submissions and feedback.
Closes https://github.com/ls1intum/Artemis/issues/3847

### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Exam in which the student has participated

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exams
4. Navigate to the Exam Detail Page
5. Reset the Exam
6. Check if the exam has been reset 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





 exam-management.service.ts: 54.17% | 95.06%  
exam-detail.component.ts: 75% |  97.22%
ExamService.java: 75% | 91% 
ExamResource: 87% | 95% 

### Screenshots



<img width="463" alt="Bildschirmfoto 2021-10-18 um 19 16 13" src="https://user-images.githubusercontent.com/51987021/137786751-fcddf23f-269c-4801-a47e-a7748b9157f1.png">
<img width="812" alt="Bildschirmfoto 2021-10-17 um 20 38 45" src="https://user-images.githubusercontent.com/51987021/137640562-7fa7dbe3-1405-4fbb-acff-6bd7b9ddff64.png">
